add core dev 2023-09 (#276)
bitcointranscripts/bitcointranscripts: bitcoin-core-dev-tech/2023-09/p2p-design-goals.md

---
title: P2P Design Goals
categories: ['core-dev-tech']
tags: ['bitcoin core', 'p2p']
date: 2023-09-20
---

## Guiding Questions

- _What are we trying to achieve?_
- _What are we trying to prevent?_
- _How so we weight performance over privacy?_
- _What is our tolerance level for net attacks?_

- Are we trying to add stuff to the network or are we trying to prevent people getting information?
  - Network topology: By design we are trying to prevent the topology being known

- Information creation, addresses, txs or blocks
  - We want blocks at tips fast - consensus critical information needs to be as fast as possible - ability to get the information - forgetting that there are multiple networks - the purpose is to know what the current most work chain tip is
    - Even the addr relay network - everything falls apart if you can’t get that information

>     - The crux of the p2p network it to propagate what is the most work

Reviewer: ```suggestion
    - The crux of the p2p network is to propagate what is the most work
```

>   - Txs can take much much longer bt don’t want to leak the source of the origin

Reviewer: ```suggestion
  - Txs can take much much longer but don’t want to leak the source of the origin
```

Could give up fast propagations for privacy for txs that would be fine but not for blocks. Speed is the most important factor.

More difficult to censor transactions because we have this decentralized network

Want a transaction to get to a miner

If there is an attacker which censors a transaction so that it is not propagated is a bigger concern. So the number 1 concern for a tx is censorship resistance then.

This is where different kind of actors come into play. If you try to stop the block propagation then

The pools are running bitcoin core nodes - fast connection to get to send in the blockshares and get the blocktemplate but they connect to pools

Censoring the chaintip is a big deal but not having your tx in my mempool isn’t a huge problem

- addrRelay exists to support these goals
  - The goal of addr is decentralization
  - Addr message may reveal about the network topology for example

Censorship resistance - create a network where everyone where connects and passes around information

- Your node doesn’t know if you are a reachable node. The network will tell you if it is reachable. From a network perspective:
  - Is providing inbounds purely altruistic

>   - From a personal standpoint, are there any advantages to accepting in bounds?

Reviewer: ```suggestion
  - From a personal standpoint, are there any advantages to accepting inbounds?
```
